File: salesos/frontend/src/application/dashboard/dashboard.dto.ts
import type { DashboardWidget } from './widget.contract'

export interface DashboardDTO {
  generatedAt: string | null
  period: 'today' | 'week' | 'month' | 'quarter'
  totalTracked: number

  missionCenter: DashboardWidget<MissionCenterData> | null
  decisionQueue: DashboardWidget<DecisionQueueData> | null
  intelligenceFeed: DashboardWidget<IntelligenceFeedData> | null
  aiBrief: DashboardWidget<AIBriefData> | null
  marketPulse: DashboardWidget<MarketPulseData> | null
  recentActivity: DashboardWidget<RecentActivityData> | null
}

export interface MissionCenterData {
  companiesTracked: number
  activeDeals: number
  pipelineValue: number
  signalsToday: number
  decisionsPending: number
}

export interface DecisionItem {
  id: string
  companyId: string
  companyName: string
  type: 'opportunity' | 'risk' | 'recommendation'
  title: string
  priority: 'high' | 'medium' | 'low'
  dueBy?: string
  score: number
}

export interface DecisionQueueData {
  items: DecisionItem[]
  total: number
}

export interface SignalItem {
  id: string
  companyId: string
  companyName: string
  category: 'tender' | 'regulatory' | 'competitor' | 'financial' | 'news'
  title: string
  summary: string
  severity: 'low' | 'medium' | 'high' | 'critical'
  source: string
  timestamp: string
  isUnseen: boolean
}

export interface IntelligenceFeedData {
  items: SignalItem[]
  total: number
  unseenCount: number
}

export interface AIBriefData {
  summary: string
  highlights: string[]
  generatedAt: string
}

export interface MarketTrend {
  name: string
  direction: 'up' | 'down' | 'stable'
  change: number
  description: string
}

export interface CompanyMover {
  companyId: string
  companyName: string
  scoreChange: number
  reason: string
}

export interface MarketPulseData {
  trends: MarketTrend[]
  topMovers: CompanyMover[]
}

export interface ActivityItem {
  id: string
  type: 'signal' | 'decision' | 'update' | 'note'
  title: string
  companyId?: string
  companyName?: string
  timestamp: string
}

export interface RecentActivityData {
  items: ActivityItem[]
  total: number
}

File: salesos/backend/app/application/dashboard/dto/dashboard_dto.py
from datetime import datetime
from pydantic import BaseModel

from app.application.dashboard.dto.widget_contract import DashboardWidget


class MissionCenterData(BaseModel):
    companiesTracked: int = 0
    activeDeals: int = 0
    pipelineValue: float = 0.0
    signalsToday: int = 0
    decisionsPending: int = 0


class DecisionItem(BaseModel):
    id: str
    companyId: str
    companyName: str
    type: str  # 'opportunity' | 'risk' | 'recommendation'
    title: str
    priority: str  # 'high' | 'medium' | 'low'
    dueBy: str | None = None
    score: float = 0.0


class DecisionQueueData(BaseModel):
    items: list[DecisionItem] = []
    total: int = 0


class SignalItem(BaseModel):
    id: str
    companyId: str
    companyName: str
    category: str  # 'tender' | 'regulatory' | 'competitor' | 'financial' | 'news'
    title: str
    summary: str
    severity: str  # 'low' | 'medium' | 'high' | 'critical'
    source: str
    timestamp: str
    isUnseen: bool = True


class IntelligenceFeedData(BaseModel):
    items: list[SignalItem] = []
    total: int = 0
    unseenCount: int = 0


class AIBriefData(BaseModel):
    summary: str = ""
    highlights: list[str] = []
    generatedAt: str = ""


class MarketTrend(BaseModel):
    name: str
    direction: str  # 'up' | 'down' | 'stable'
    change: float = 0.0
    description: str = ""


class CompanyMover(BaseModel):
    companyId: str
    companyName: str
    scoreChange: float = 0.0
    reason: str = ""


class MarketPulseData(BaseModel):
    trends: list[MarketTrend] = []
    topMovers: list[CompanyMover] = []


class ActivityItem(BaseModel):
    id: str
    type: str  # 'signal' | 'decision' | 'update' | 'note'
    title: str
    companyId: str | None = None
    companyName: str | None = None
    timestamp: str


class RecentActivityData(BaseModel):
    items: list[ActivityItem] = []
    total: int = 0


class DashboardDTO(BaseModel):
    generatedAt: datetime | None = None
    period: str = "today"
    totalTracked: int = 0

    missionCenter: DashboardWidget | None = None
    decisionQueue: DashboardWidget | None = None
    intelligenceFeed: DashboardWidget | None = None
    aiBrief: DashboardWidget | None = None
    marketPulse: DashboardWidget | None = None
    recentActivity: DashboardWidget | None = None

    scoredDecisions: list[dict] = []
    overallHealth: float = 1.0
